### PR TITLE
feat(rest-api): Adds Swagger for api docs [SLT-205]

### PR DESCRIPTION
**Pull request by abtestingalpha**
**Description**
* Adds Swagger for `/api-docs`
* Initial docs with `/destinationTokens` endpoint

## Summary by CodeRabbit

- **New Features**
	- Integrated Swagger UI for enhanced API documentation, accessible at `/api-docs`.
	- Introduced a new endpoint to retrieve destination tokens based on source chain ID and token address.
	- Added OpenAPI documentation for the `/bridge` endpoint to retrieve quotes for bridging tokens.
	- Added OpenAPI documentation for the `/synapseTxId` endpoint to retrieve Synapse transaction IDs.
	- Added OpenAPI documentation for the `/swap` endpoint to provide detailed swap quotes.
	- Added a new endpoint to retrieve swap transaction information.
	- Added a new endpoint to retrieve bridge transaction status.

- **Improvements**
	- Added new dependencies to support Swagger documentation features.
	- Disabled strict indentation checks for JSDoc comments, allowing more flexible formatting.

These changes enhance usability and provide developers with interactive documentation for better API integration.

### CODE REVIEW

**Comment by Defi-Moses** [comment]
Some non-critical ideas:

How do we feel about having users land on the api-docs page instead of the get api information? Which is more helpful to users?

Also should we rename the tx status endpoints to remove the "get" ie:
`getSynapseTxId` -> `synapseTxId`
`getBridgeTxStatus` -> `bridgeTxStatus`
`getDestinationTx` -> `destinationTx`

otherwise looks good to me

**Review by Defi-Moses — APPROVED** [review]

File: packages/rest-api/src/routes/swapTxInfoRoute.ts [hunk 1]
```diff
@@ -11,6 +11,108 @@
...
+ *                   description: The address of the contract to interact with
+ *                 value:
+ *                   type: object
+ *                   properties:
+ *                     type:
+ *                       type: string
+ *                       enum: [BigNumber]
+ *                     hex:
+ *                       type: string
+ *                   description: The amount of native currency to send with the transaction
```
[the remark this answers is not in the document]
Comment (abtestingalpha) [reply packages/rest-api/src/routes/swapTxInfoRoute.ts:73]: Can't do this bc this is what the sdk responds with

File: packages/rest-api/src/routes/bridgeTxInfoRoute.ts [hunk 2]
```diff
@@ -11,6 +11,116 @@
...
+ *                     description: The address of the contract to interact with
+ *                   value:
+ *                     type: object
+ *                     properties:
+ *                       type:
+ *                         type: string
+ *                         enum: [BigNumber]
+ *                       hex:
+ *                         type: string
+ *                     description: The amount of native currency to send with the transaction
```
[the remark this answers is not in the document]
Comment (abtestingalpha) [reply packages/rest-api/src/routes/bridgeTxInfoRoute.ts:81]: Can't do this bc this is how the sdk response

File: packages/rest-api/src/routes/bridgeRoute.ts [hunk 3]
```diff
@@ -10,6 +10,181 @@
...
+ *                     type: string
+ *                   feeAmount:
+ *                     $ref: '#/components/schemas/BigNumber'
+ *                   feeConfig:
+ *                     type: object
+ *                     properties:
+ *                       bridgeFee:
+ *                         type: integer
+ *                       minFee:
+ *                         $ref: '#/components/schemas/BigNumber'
```
[the remark this answers is not in the document]
Comment (abtestingalpha) [reply packages/rest-api/src/routes/bridgeRoute.ts:70]: It is an integer though

File: packages/rest-api/src/routes/bridgeRoute.ts [hunk 4]
```diff
@@ -10,6 +10,181 @@ import { checksumAddresses } from '../middleware/checksumAddresses'
...
+ *                   maxFee:
+ *                     type: "BigNumber"
+ *                     hex: "0x17d78400"
+ *                 routerAddress: "0xd5a597d6e7ddf373a92C8f477DAAA673b0902F48"
+ *                 maxAmountOut:
+ *                   type: "BigNumber"
+ *                   hex: "0xe89bd2cb27"
+ *                 originQuery:
+ *                   routerAdapter: "0x0000000000000000000000000000000000000000"
+ *                   tokenOut: "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48"
```
[the remark this answers is not in the document]
Comment (abtestingalpha) [reply packages/rest-api/src/routes/bridgeRoute.ts:114]: Not sensitive, an example of output of a public query

**Pull request merged**